ls1intum/Orion
# [Feature] Assessment in Orion

### Motivation and Context
Enables editing inline comments in Orion. Requires https://github.com/ls1intum/Artemis/pull/3913.

### Description
Adds a new OrionAssessmentService which handles the feedback internally and an OrionEditorProvider to provide the OrionAssessmentEditor with the ability to edit the assessment.

### Limitations
The following features/problems will be adressed in a followup
 - Assessing Feedback requests, complaints and exams
 - The web client layout does currently not deal with the small Orion window well
 - There is a strange bug that prevents certain key events (like enter, arrow keys) from reaching the inline comments

### Steps for Testing

1.  Install the release as described in [the readme](https://github.com/ls1intum/Orion/blob/master/README.md#testing-of-pull-requests)
2.  Deploy the Artemis PR on a testserver and set your Orion to connect to the testserver
3.  Create/Locate a programming exercise with submissions
4.  Assess the exercise in Orion
5.  Edit the assessment in various ways
6.  Edit the code and execute it locally
7.  Save/Submit the assessment
8.  Confirm the assessment was saved correctly by checking it in the web client
9.  Assess another submission

### Review Progress
- Code Review
  - (Not applicable for Kotlin)
- Manual Tests

### Screenshots
![grafik](https://user-images.githubusercontent.com/61357727/130869240-254ae4bf-a3ee-4de1-b87e-2d9c050ae45e.png)
Orion's Assessment view with a negative, neutral and positive comment. Switch between edit and assessment mode with the tabs at the bottom of the editor. All relevant Artemis data is shown in the integrated browser.
![grafik](https://user-images.githubusercontent.com/61357727/131865002-6fbc5b4d-a366-40a5-a858-3579f2937ebe.png)
Edit mode in light mode
